web/app/js/components/Navigation.jsx

```diff
@@ -356,7 +364,11 @@ class NavigationBase extends React.Component {
 
   render() {
     const { api, classes, selectedNamespace, ChildComponent, ...otherProps } = this.props;
-    const { namespaces, anchorEl, showNamespaceChangeDialog, newNamespace, mobileSidebarOpen } = this.state;
+    let { namespaces, namespaceFilter, anchorEl, showNamespaceChangeDialog,
+      newNamespace, mobileSidebarOpen } = this.state;
+    namespaces = namespaces.filter(ns => {
+      return ns.name.match(regexFilterString(namespaceFilter));
```

Comment:
I think it would be better to avoid calling `regexFilterString` into the `filter` method:

```
const formattedNamespaceFilter = regexFilterString(namespaceFilter);
namespaces = namespaces.filter(ns => {
    return ns.name.match(formattedNamespaceFilter);
});
```

... or much better, call `regexFilterString` only to change the input and save the `formatted` namespaceFilter version into the state. This way we avoid building the regular expression multiple times.

```
handleFilterInputChange = event => {
    this.setState({ namespaceFilter: event.target.value,  formattedNamespaceFilter: regexFilterString(event.target.value)});
}
```

Reply:
Thanks for the idea! Having a separate `formattedNamespace` variable was actually part of my original PR but I omitted it after some review feedback. I like your solution of calculating the regex when saving it into the state, and have restored the variable. 👍